metricbeat/module/jolokia/jmx/jmx.go

```diff
@@ -75,12 +78,18 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 
 	log := logp.NewLogger(metricsetName).With("host", base.HostData().Host)
 
+	http, err := helper.NewHTTP(base)
+	if err != nil {
+		return nil, err
+	}
+
 	return &MetricSet{
-		BaseMetricSet: base,
-		mapping:       config.Mappings,
-		namespace:     config.Namespace,
-		http:          jolokiaHTTPBuild,
-		log:           log,
+		BaseMetricSet:    base,
+		mapping:          config.Mappings,
+		namespace:        config.Namespace,
+		jolokiaHTTPBuild: jolokiaHTTPBuild,
```

Comment:
We could maybe rename this to something like:
```suggestion
		jolokia: jolokiaFetcher,
```

The `...Build` name is a bit weird here, I think this comes from the original implementation.

Reply:
Yes, indeed. Renamed.